crates/ruff/src/rules/flake8_bugbear/rules/mutable_argument_default.rs

```diff
@@ -78,9 +88,37 @@ pub(crate) fn mutable_argument_default(checker: &mut Checker, arguments: &Argume
                 is_immutable_annotation(expr, checker.semantic())
             })
         {
-            checker
-                .diagnostics
-                .push(Diagnostic::new(MutableArgumentDefault, default.range()));
+            let mut diagnostic = Diagnostic::new(MutableArgumentDefault, default.range());
+
+            if checker.patch(diagnostic.kind.rule()) {
+                // Set the default arg value to None
+                let arg_edit = Edit::range_replacement("None".to_string(), default.range());
+
+                // Add conditional check to set the default arg to its original value if still None
+                let mut check_lines = String::new();
+                let indentation = checker.locator.slice(TextRange::new(
+                    checker.locator.line_start(body[0].start()),
+                    body[0].start(),
+                ));
+                let indentation = leading_space(indentation);
+                check_lines.push_str(format!("if {} is None:\n", def.arg.as_str()).as_str());
```

Comment:
i was confused at first why there was no indentation before the `if`, would make sense to add a comment about it